lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs

```diff
@@ -285,10 +287,10 @@ instance FromJSON SlotNo where
     parseJSON = fmap SlotNo . parseJSON
 
 instance ToJSON EpochNo where
-    toJSON (EpochNo n) = toJSON n
+    toJSON (EpochNo n) = toJSON (fromIntegral @Word31 @Word32 n)
 
 instance FromJSON EpochNo where
-    parseJSON = fmap EpochNo . parseJSON
+    parseJSON = fmap (EpochNo . fromIntegral @Word32 @Word31) . parseJSON
```

Comment:
We probably want to fail loudly here too.

Reply:
Hmm No. These instances are not used and are "unsound". Sadly required by the persistent auto-generated machinery for a mysterious reason :disappointed: